### PR TITLE
fix(editor): Fix local storage flags defaulting to undefined string

### PR DESCRIPTION
**Pull request by mutdmour**
useStorage takes the default value `undefined` and sets it in local storage.. also returns "undefined" as string which breaks onboarding flows

Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

**Review by netroy — COMMENTED** [review]
One major advantage of `useStorage` composable is that it's reactive and synchronized across tabs/windows automatically.
Perhaps we can create our own `useStorage` composable instead?

**Review by cstuncsik — DISMISSED** [review]

File: packages/editor-ui/src/components/ActivationModal.vue [hunk 1]
```diff
@@ -88,7 +89,7 @@ export default defineComponent({
 		},
 		handleCheckboxChange(checkboxValue: boolean) {
 			this.checked = checkboxValue;
-			window.localStorage.setItem(LOCAL_STORAGE_ACTIVATION_FLAG, checkboxValue.toString());
+			useStorage(LOCAL_STORAGE_ACTIVATION_FLAG).value = checkboxValue.toString();
```
Comment (netroy) [inline packages/editor-ui/src/components/ActivationModal.vue:92]: same here. we could add the ref in `setup`
```typescript
	setup() {
		return {
			activationFlag: useStorage(LOCAL_STORAGE_ACTIVATION_FLAG),
		};
	},
```

and then update this to 
```suggestion
			this.activationFlag.value = checkboxValue.toString();
```

File: packages/editor-ui/src/composables/useStorage.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+		// bug in 1.15.1 where it returns 'undefined' as a string
+		if (storedValue !== null && storedValue !== 'undefined') {
+			data.value = storedValue;
+		}
+	} catch (e) {
+		console.log(`Failed to read from localStorage for key ${key}`);
+	}
+
+	// Watch the ref and update localStorage whenever it changes
+	watchEffect(() => updateLocalStorage(data.value));
```
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/editor-ui/src/composables/useStorage.ts]: update to simply wrap the original composable

File: packages/editor-ui/src/stores/ui.utils.ts [hunk 3]
```diff
@@ -11,7 +12,7 @@ export function isValidTheme(theme: string | null): theme is AppliedThemeOption
 
 // query param allows overriding theme for demo view in preview iframe without flickering
 export function getThemeOverride() {
-	return getQueryParam('theme') || localStorage.getItem(LOCAL_STORAGE_THEME);
+	return getQueryParam('theme') || useStorage(LOCAL_STORAGE_THEME).value;
```
[the remark this answers is not in the document]
Comment (mutdmour) [reply packages/editor-ui/src/stores/ui.utils.ts]: updated.. I would rather we add the refs to store to get the benefits of reactivity.. but too much to refactor and test this now..

File: packages/editor-ui/src/components/ActivationModal.vue (hunk 1 above)
Comment (mutdmour) [reply packages/editor-ui/src/components/ActivationModal.vue:92]: since it's not reused, will leave as is for now.

**Review by netroy — DISMISSED** [review]
🙏🏽

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.16.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.16.0)

**Pull request merged**